File: app/src/main/java/com/pr0gramm/app/ui/intro/slides/SettingActionItem.java
package com.pr0gramm.app.ui.intro.slides;

import com.pr0gramm.app.Settings;

/**
 */
class SettingActionItem extends ActionItem {
    private final Settings settings;
    private final String preference;

    SettingActionItem(Settings settings, String title, String preference) {
        super(title);
        this.settings = settings;
        this.preference = preference;
    }

    @Override
    public boolean enabled() {
        return settings.raw().getBoolean(preference, false);
    }

    @Override
    public void activate() {
        settings.edit().putBoolean(preference, true).apply();
    }

    @Override
    public void deactivate() {
        settings.edit().putBoolean(preference, false).apply();
    }
}

File: app/src/main/java/com/pr0gramm/app/ui/intro/slides/SettingActionItem.kt
package com.pr0gramm.app.ui.intro.slides

import com.pr0gramm.app.Settings

/**
 * Configures settings.
 */
internal class SettingActionItem(title: String, private val preference: String) : ActionItem(title) {
    override fun enabled(): Boolean {
        return Settings.raw().getBoolean(preference, false)
    }

    override fun activate() {
        Settings.edit {
            putBoolean(preference, true)
        }
    }

    override fun deactivate() {
        Settings.edit {
            putBoolean(preference, false)
        }
    }
}
